### PR TITLE
Fix removing temporary branch on --check-github

### LINKED ISSUE #3181: "Failed to delete test branch from GitHub" when using --check-github
When setting EASYBUILD_GIT_WORKING_DIRS_PATH to my local repo I get the following error:

```
* push access to Flamefire/easybuild-easyconfigs repo @ GitHub...OK
WARNNIG: failed to delete test branch from GitHub: Cmd('git') failed due to: exit code(1)
  cmdline: git push --delete --porcelain origin test_branch_ZOSYM
  stderr: 'error: unable to delete 'test_branch_ZOSYM': remote ref does not exist
error: failed to push some refs to '/home/s3248973/.local/EasyBuildDev/easybuild-easyconfigs/.git''
```

When I go to the mentioned folder (without the .git) and execute the command it does work

### PR DESCRIPTION
**Pull request by Flamefire**
The remote can be somethings else than "origin" so make sure we take the one used to create the branch

Also fixes the silent output even when passing --debug

Fixes #3181

### CODE REVIEW

**Comment by Flamefire** [comment]
@boegel This bug was introduced in https://github.com/easybuilders/easybuild-framework/commit/b07a54cca9ca219087c588053368e74cbb74ff0a and should have existed (and happened) always since then

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -1726,10 +1728,9 @@ def check_github():
...
 
     # cleanup: delete test branch that was pushed to GitHub
-    if git_repo:
+    if git_repo and push_err is None:
         try:
-            if git_repo and hasattr(git_repo, 'remotes') and hasattr(git_repo.remotes, 'origin'):
-                git_repo.remotes.origin.push(branch_name, delete=True)
+            getattr(git_repo.remotes, remote_name).push(branch_name, delete=True)
         except GitCommandError as err:
             sys.stderr.write("WARNNIG: failed to delete test branch from GitHub: %s\n" % err)
```
Comment (ocaisa) [inline easybuild/tools/github.py]: Might as well fix typo while we are here: `WARNING`

**Comment by Flamefire** [comment]
> Can we get a test to capture this bug?

I'd rather not. It is either a lot of work to mock everything out or have changes being done to the real github repo which would even require a token.

**Comment by Flamefire** [comment]
Funny: The github tests are not run the `--check-github` is even failing:

 ```
>> python -m easybuild.main --check-github --github-user=easybuild_test
 
 WARNING: Robot search path is empty!
 
 == temporary log file in case of crash /tmp/eb-ajOiTU/easybuild-PYOUBL.log
 
 Checking status of GitHub integration...
 
 Making sure we're online...OK
 
 * GitHub user...easybuild_test => OK
 * GitHub token...(no token found) => FAIL
 * git command...OK ("git version 2.25.0; ")
 * GitPython module...OK (GitPython version 2.1.14)
 * push access to easybuild_test/easybuild-easyconfigs repo @ GitHub...FAIL (unexpected exception: 'Checking out branch \'master\' from git@github.com:easybuild_test/easybuild-easyconfigs.git failed: "Failed to fetch branch \'master\' from git@github.com:easybuild_test/easybuild-easyconfigs.git: Cmd(\'git\') failed due to: exit code(128)\\n  cmdline: git fetch -v pr_target_account_easybuild_test_qsbQC\\n  stderr: \'fatal: Could not read from remote repository.\\n\\nPlease make sure you have the correct access rights\\nand the repository exists.\'"')
 * creating gists...FAIL (gist_url: None)
 * location to Git working dirs... not found (suboptimal)
 
 One or more checks FAILed, GitHub configuration not fully complete!
 See http://easybuild.readthedocs.org/en/latest/Integration_with_GitHub.html#configuration for help.
 
 Status of GitHub integration:
 * --from-pr: OK
 * --new-pr: not supported
 * --review-pr: OK
 * --update-pr: not supported
 * --upload-test-report: not supported
 
 ...................................Skipping test_empty_pr, no GitHub token available?

```

**Comment by Flamefire** [comment]
Rebased to dev to include the timeout fix and make CI pass. Please merge, I'm sick of those tmp_* branches ;)

**Pull request merged**